<!---
This is a dynamically generated file. Do not edit manually.
date:        Sat, 24 Aug 2019 01:45:03 GMT
author:      "Peter Chung <touhonoob@gmail.com>"
--->

# no-mix-tabs-and-spaces
![Recommended Badge](https://img.shields.io/badge/-Recommended-brightgreen)
![Category Badge](https://img.shields.io/badge/-Style%20Guide%20Rules-informational)
![Default Severity Badge warn](https://img.shields.io/badge/Default%20Severity-warn-yellow)
> The {"extends": "solhint:recommended"} property in a configuration file enables this rule.


## Description
Mixed tabs and spaces.

## Options
This rule does not have options.

## Examples
### ðŸ‘Ž Examples of **incorrect** code for this rule

#### Expression with mixed tabs and spaces

```solidity
 	 import "lib.sol";
```

## Version
This rule was introduced in [Solhint 1.1.5](https://github.com/protofire/solhint/tree/v1.1.5)

## Resources
- [Rule source](https://github.com/protofire/solhint/tree/master/lib/rules/align/no-mix-tabs-and-spaces.js)
- [Document source](https://github.com/protofire/solhint/tree/master/docs/rules/align/no-mix-tabs-and-spaces.md)
- [Test cases](https://github.com/protofire/solhint/tree/master/test/rules/align/no-mix-tabs-and-spaces.js)
